plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/connection/ConnectionWizardPageModel.java

```diff
@@ -42,95 +46,114 @@
-		}
+	private ConnectionsFactoryTracker createConnectionsFactory() {
+		ConnectionsFactoryTracker connectionsFactory = new ConnectionsFactoryTracker();
+		connectionsFactory.open();
+		return connectionsFactory;
 	}
 
-	protected String getDefaultUsername() {
-		return "";
-//		String username = ExpressPreferences.INSTANCE.getLastUsername();
-//		if (StringUtils.isEmpty(username)) {
-//			try {
-//				username = new OpenShiftConfiguration().getRhlogin();
-//			} catch (IOException e) {
-//				OpenShiftCommonUIActivator.log("Could not load default user name from OpenShift configuration.", e);
-//			} catch (OpenShiftException e) {
-//				OpenShiftCommonUIActivator.log("Could not load default user name from OpenShift configuration.", e);
-//			}
-//		}
-//		return username;
+	private void updateModel(IConnection connection, IConnectionFactory factory, IStatus connectionCreationError, IStatus connectError) {
+		updateConnectionFactory(factory);
+		updateConnection(connection, connectionCreationError, connectError);
 	}
 
-	private boolean isNewConnection(IConnection connection) {
-		return connection instanceof NewConnectionMarker;
+	private void updateConnectionFactory(IConnectionFactory factory) {
+		if (factory != null
+				&& !factory.equals(connectionFactory)) {
+			if (isUseDefaultHost()) {
+				setUseDefaultHost(factory.hasDefaultHost(), false);
+				if (!StringUtils.isEmpty(factory.getDefaultHost())) {
+					setHost(factory.getDefaultHost(), false);
+				}
+			}
+		}
 	}
 
-	private void resetConnection(IStatus status) {
-		setConnection(null, status);
+	private void updateConnection(IConnection connection, IStatus connectionCreationError, IStatus connectError) {
+		if (!equalsTypeAndHost(connection, this.connection)) {
+			if (connection != null) {
+				// switch from Automatic->specific server type
+				setConnectionFactory(connectionsFactory.getByConnection(connection.getClass()), false);
+			}
+			setConnection(connection, false);
+		}
+		setConnectionCreationError(connectionCreationError);
+		setConnectError(connectError);
 	}
 
 	public void setConnection(IConnection connection) {
-		setConnection(connection, ValidationStatus.ok());
+		setConnection(connection, true);
 	}
-	
-	private void setConnection(IConnection connection, IStatus connectionCreationStatus) {
-//		if (Diffs.equals(this.connection, connection)) {
-//			return;
-//		}
-		if (connection != null) {
-			setConnectionFactory(connectionsFactory.getByConnection(connection.getClass()));
+
+	private void setConnection(IConnection connection, boolean updateModel) {
+		if (updateModel) {
+			if (connection != null) {
+				updateModel(connection, connectionFactory, Status.OK_STATUS, Status.OK_STATUS);
+			} else {
+				updateModel(connection, connectionFactory, Status.CANCEL_STATUS, Status.OK_STATUS);
+			}
 		}
-		setConnectionCreated(connectionCreationStatus);
-		updateFrom(connection);
+		addConnectionListener(connection, this.connection);
```

Comment:
@adietish is there any concern here that if 'updateModel' = false that the connection might be null?  and if we are updating is there a need to add a connection listener?